### DISCUSSION TITLE
Questions on Computational Project--Wavefunction Collapse Algorithm

### DISCUSSION THREAD

**Discussion by WilsonFangCam** [Part-II-Computational-Physics/.github#78] [Q&A]
While I was trying the python implementation linked to the description of the project of the Wavefunction Collapse Algorithm, I found that implementation very very complicated and contains very little documentation. I am not even sure how to execute this program on my computer.

<img width="878" alt="image" src="https://user-images.githubusercontent.com/125758929/225178034-2763dc07-c040-42e5-9dd9-c4100a25f5fa.png">

How am I supposed to use it?

Should I try to copy (with citations) some parts of his code as a starting point for developing a well-performed program?

Or, should I write my own scripts from the very beginning and create my own implementation, which probably wouldn't perform very well?

Like, can I get some instructions on roughly the percentage of lines of codes that I should write on my own vs. quoting from others?

**Comment by AustenLamacraft** [comment]
After installing the requirements in `requirements.txt` with 

`pip install -r requirements.txt`

I found the command `python wfc_run.py -s samples_original.xml` produces output in `/output`. Note that `wfc_run.py` is principally a script to run the main algo implemented in `/wfc`.

A more lightweight Python implementation is described here:

https://terbium.io/2018/11/wave-function-collapse/

which has as an accompanying [notebook](https://terbium.io/2018/11/wave-function-collapse/wave-function-collapse.ipynb).
